### ISSUE TITLE
Question about deformation updates in Eqs. 5-7 of the arXiv paper

### ISSUE THREAD

**Issue by xiaogengen-007** [honkamj/SITReg#1]
Hi @@honkamj,

Thank you so much for this great work and making the code publicly available!

I have a question regarding the Eqs. 5-7 in the [arxiv](https://arxiv.org/html/2303.10211v5) paper. If I understand correctly, the update deformation δ(k) in Eq. 5 is a full-way deformation between the warped feature maps z1 and z2, instead of half-way deformations, which are u(z1, z2) and u(z2, z1). If that interpretation is correct, should the half-way updates in Eqs. 6-7 use u(z1, z2) and u(z2, z1) instead of δ(k)?

Thank you again for your time and effort in sharing this work!

**Comment by honkamj** [comment]
Hi! Thank you for the question. The update deformation δ(k) is only half-way deformation, as written in the paper. The update deformation itself is predicted from u(z1, z2) and u(z2, z1) to make prediction (anti)symmetric with respect to z1 and z2. The deformations u(z1, z2) and u(z2, z1)  do not have any clear meaning themselves individually. The δ(k) is a half-way instead of full deformation to maintain the symmetry over multiple stages.

**Comment by xiaogengen-007** [comment]
Hi!
Thanks for the clarification!

I am still a little bit confused.
I previously thought u(z1, z2) and u(z2, z1) in Eq. 5 were halfway deformations. 
I guess my misunderstanding arised from that in Eq. 2 and Eq. 3,  the symbol u was used for halfway deformation. 
```
f(xA, xB) := u(xA, xB) ◦ u(xB, xA)−1
f(xA, xB) := u(h(xA), h(xB)) ◦ u(h(xB), h(xA))−1
```
Are the halfway deformation further broken down into 1/4 deformations (or something that do not have any clear meaning) to enforce anti-symmetry?

Thank you!

**Comment by honkamj** [comment]
Hi again!

Oh yes, initially in Eq 2 the u(xA, xB) ◦ u(xB, xA)−1 describes the whole deformation, as the idea is to describe the basic idea for building anti-symmetric functions. Note also that u itself is not a symbol for deformation here, it is a neural network outputting a deformation, and u(xA, xB), u(xB, xA) are just outputs of the same network with different input orders.

However, even there u(xA, xB) and u(xB, xA) are not necessarily half-way deformations since when training a network only the composition is enforced to approach something. Hence the u(xA, xB) and u(xB, xA) might divide the whole deformation a priori in arbitrary manner (in practice they do probably learn to be approximately half-way deformations if trained with symmetric loss).

In summary, in symmetric multi-stage architecture we want to learn half-way (square root) deformations for the symmetry to extend across stages. And those half-way deformations still have to be predicted via an anti-symmetric function. Hence the formulation (composition of four deformations) for the whole deformation. I hope this clarifies!

**Comment by xiaogengen-007** [comment]
Thank you for the thorough explanation! That is very helpful!
